src/Parser/LRUCache.php

```diff
@@ -0,0 +1,72 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Parser;
+
+use LogicException;
+use function array_key_first;
+use function array_keys;
+use function count;
+use function sprintf;
+
+/**
+ * @template TValue
+ * @implements ParserCache<TValue>
+ */
+class LRUCache implements ParserCache
+{
+
+	/** @var array<string, TValue> */
+	private array $cache;
```

Comment:
it might make sense to try SplFixedArray here

Reply:
Hmm our capacity is too small for it beeing useful

https://stackoverflow.com/a/11831076